pkg/plugin/healthz.go

```diff
@@ -84,21 +94,73 @@ func (h *HealthZ) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
 		http.Error(w, "plain text mismatch after decryption", http.StatusInternalServerError)
 		return
 	}
+
+	// v2 checks
+	// generate UID
+	uid, err := utils.GenerateUID()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	v2EncryptResponse, err := h.KMSV2Server.Encrypt(
+		ctx,
+		&kmsv2.EncryptRequest{
+			Plaintext: []byte(healthCheckPlainText),
+			Uid:       uid,
+		},
+	)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	v2DecryptResponse, err := h.KMSV2Server.Decrypt(ctx, &kmsv2.DecryptRequest{
+		Ciphertext:  v2EncryptResponse.Ciphertext,
+		KeyId:       v2EncryptResponse.KeyId,
+		Uid:         uid, // passing the same uid to track roundtrip encrypt/decrypt calls
+		Annotations: v2EncryptResponse.Annotations,
+	})
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if string(v2DecryptResponse.Plaintext) != healthCheckPlainText {
+		http.Error(w, "plain text mismatch after decryption", http.StatusInternalServerError)
+		return
+	}
```

Comment:
Lets add a comment for
1. how many additional kv calls this results in (`v1` + `v2`)
2. how frequently this check is done